semantickernel-experimental/src/main/java/com/microsoft/semantickernel/connectors/data/postgres/PostgreSQLVectorStoreQueryProvider.java

```diff
@@ -0,0 +1,302 @@
+    private PostgreSQLVectorStoreQueryProvider(DataSource dataSource, String collectionsTable,
+        String prefixForCollectionTables) {
+        super(dataSource, collectionsTable, prefixForCollectionTables);
+        this.dataSource = dataSource;
+        this.collectionsTable = collectionsTable;
+        this.prefixForCollectionTables = prefixForCollectionTables;
+
+        supportedKeyTypes = new HashMap<>();
+        supportedKeyTypes.put(String.class, "VARCHAR(255)");
+
+        supportedDataTypes = new HashMap<>();
+        supportedDataTypes.put(String.class, "TEXT");
+        supportedDataTypes.put(Integer.class, "INTEGER");
+        supportedDataTypes.put(int.class, "INTEGER");
+        supportedDataTypes.put(Long.class, "BIGINT");
+        supportedDataTypes.put(long.class, "BIGINT");
+        supportedDataTypes.put(Float.class, "REAL");
+        supportedDataTypes.put(float.class, "REAL");
+        supportedDataTypes.put(Double.class, "DOUBLE PRECISION");
+        supportedDataTypes.put(double.class, "DOUBLE PRECISION");
+        supportedDataTypes.put(Boolean.class, "BOOLEAN");
+        supportedDataTypes.put(boolean.class, "BOOLEAN");
+        supportedDataTypes.put(OffsetDateTime.class, "TIMESTAMPTZ");
+
+        supportedVectorTypes = new HashMap<>();
+        supportedDataTypes.put(String.class, "TEXT");
+        supportedVectorTypes.put(List.class, "VECTOR(%d)");
+        supportedVectorTypes.put(Collection.class, "VECTOR(%d)");
+    }
+
+    /**
+     * Creates a new builder.
+     * @return the builder
+     */
+    public static PostgreSQLVectorStoreQueryProvider.Builder builder() {
+        return new PostgreSQLVectorStoreQueryProvider.Builder();
+    }
+
+    /**
+     * Prepares the vector store.
+     * Executes any necessary setup steps for the vector store.
+     *
+     * @throws SKException if an error occurs while preparing the vector store
+     */
+    @Override
+    public void prepareVectorStore() {
+        super.prepareVectorStore();
+
+        // Create the vector extension
+        String pgVector = "CREATE EXTENSION IF NOT EXISTS vector";
+
+        try (Connection connection = dataSource.getConnection();
+            PreparedStatement createPgVector = connection.prepareStatement(pgVector)) {
+            createPgVector.execute();
+        } catch (SQLException e) {
+            throw new SKException("Failed to prepare vector store", e);
+        }
+    }
+
+    private String getColumnNamesAndTypesForVectorFields(List<VectorStoreRecordVectorField> fields,
+        Class<?> recordClass) {
+        StringBuilder columnNames = new StringBuilder();
+        for (VectorStoreRecordVectorField field : fields) {
+            try {
+                Field declaredField = recordClass.getDeclaredField(field.getName());
+                if (columnNames.length() > 0) {
+                    columnNames.append(", ");
+                }
+
+                if (declaredField.getType().equals(String.class)) {
```

Comment:
Are you missing an `append(" ")` between the field name and vector type in this block? 

Too bad the String type had to be special cased. There's only three supported vector types (according to your code), so maybe this isn't a big deal. Just a thought, though. You could put the string formating in the hash table instead of just the string format. Make it a `Map<Class<?>, BiFunction<String, Integer, String>>` then you can `put(String.class, (fieldName, dimension) -> String.format("%s TEXT", fieldName))` and `put(List.class, (fieldName, dimension) -> String.format("%s VECTOR(%d)", fieldName, dimension))` and just do `get(declaredField.getType()).apply(field.getName(), field.getDimensions()). But again, I wouldn't change the code you have. Just fun to play with.

Reply:
Great catch!